From 4c996a57d4c01f092f9ed8f98a9f476c14c0c777 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Martin=20Fl=C3=B6ser?= <mgraesslin@kde.org>
Date: Thu, 20 Jul 2017 19:14:07 +0200
Subject: Call Platform::setupActionForGlobalAccel on the Client shortcut

Summary:
Platform::setupActionForGlobalAccel is an important call on X11 platform.
Without the x11 timestamp doesn't get updated and calls in KWin might
fail - e.g. the activation of the Client which is supposed to happen.

Test Plan: compiles

Reviewers: #kwin, #plasma

Subscribers: plasma-devel, kwin

Tags: #kwin

Differential Revision: https://phabricator.kde.org/D6802
---
 useractions.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/useractions.cpp b/useractions.cpp
index 502be96..d92b027 100644
--- a/useractions.cpp
+++ b/useractions.cpp
@@ -37,6 +37,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include "input.h"
 #include "workspace.h"
 #include "effects.h"
+#include "platform.h"
 #include "screens.h"
 #include "virtualdesktops.h"
 #include "scripting/scripting.h"
@@ -1050,6 +1051,7 @@ void Workspace::clientShortcutUpdated(Client* c)
     if (!c->shortcut().isEmpty()) {
         if (action == NULL) { // new shortcut
             action = new QAction(this);
+            kwinApp()->platform()->setupActionForGlobalAccel(action);
             action->setProperty("componentName", QStringLiteral(KWIN_NAME));
             action->setObjectName(key);
             action->setText(i18n("Activate Window (%1)", c->caption()));
-- 
cgit v0.11.2
